python/nano/test/pytorch/tests/test_inference_pipeline_ipex.py

```diff
@@ -69,25 +69,54 @@ class TestInferencePipeline(TestCase):
     trainer = Trainer(max_epochs=max_epochs)
     model = Trainer.compile(model, loss, optimizer)
     trainer.fit(model, train_loader)
+    
+    def test_get_model_without_optimize(self):
+        inference_opt = InferenceOptimizer()
+        with pytest.raises(RuntimeError) as e:
+            acc_model, option = inference_opt.get_best_model()
+        error_msg = e.value.args[0]
+        assert error_msg == "There is no optimized model. You should call .optimize() " \
+                            "before get_best_model()"
 
-    def test_pipeline(self):
+    def test_pipeline_with_metric(self):
         inference_opt = InferenceOptimizer()
-        inference_opt.optimize(model=self.model, 
+        inference_opt.optimize(model=self.model,
                                training_data=self.train_loader,
-                               validation_data=self.test_loader, 
+                               validation_data=self.test_loader,
                                metric=self.metric,
                                direction="max",
                                cpu_num=1)
-        for key, value in inference_opt.optimized_model_dict.items():
-            print(key, value["latency"], value["accuracy"])
+
         acc_model, option = inference_opt.get_best_model()
         print(option)
         acc_model, option = inference_opt.get_best_model(accelerator="onnxruntime")
         assert option == "" or "onnxruntime" in option
         acc_model, option = inference_opt.get_best_model(precision="int8")
         assert option == "" or "inc" in option
         acc_model, option = inference_opt.get_best_model(accuracy_criterion=0.1)
+        acc_model(next(iter(self.train_loader))[0])
+
+    def test_pipeline_without_metric(self):
+        inference_opt = InferenceOptimizer()
+        inference_opt.optimize(model=self.model,
+                               training_data=self.train_loader,
+                               cpu_num=1)
+
+        acc_model, option = inference_opt.get_best_model()
+        print(option)
+        acc_model, option = inference_opt.get_best_model(accelerator="onnxruntime")
+        assert option == "" or "onnxruntime" in option
+        acc_model, option = inference_opt.get_best_model(precision="int8")
+        assert option == "" or "inc" in option
```

Comment:
`assert option == "" or "inc" in option or "pot" in option`?

Reply:
Good catch! done